apis/v1alpha2/validation/grpcroute.go

```diff
@@ -63,13 +67,38 @@ func validateGRPCRouteRules(rules []gatewayv1a2.GRPCRouteRule, path *field.Path)
 	return errs
 }
 
-// validateRuleMatches validates that at least one of the fields Service or Method of
-// GRPCMethodMatch to be specified
+// validateRuleMatches validates GRPCMethodMatch
 func validateRuleMatches(matches []gatewayv1a2.GRPCRouteMatch, path *field.Path) field.ErrorList {
 	var errs field.ErrorList
 	for i, m := range matches {
-		if m.Method != nil && m.Method.Service == nil && m.Method.Method == nil {
-			errs = append(errs, field.Required(path.Index(i).Child("method"), "one or both of `service` or `method` must be specified"))
+		if m.Method != nil {
+			if m.Method.Service == nil && m.Method.Method == nil {
+				errs = append(errs, field.Required(path.Index(i).Child("method"), "one or both of `service` or `method` must be specified"))
+			}
+			// GRPCRoute method matcher admits two types: Exact and RegularExpression.
+			// If not specified, the match will be treated as type Exact
+			if m.Method.Type == nil || *m.Method.Type == gatewayv1a2.GRPCMethodMatchExact {
+				if m.Method.Service != nil {
+					r, err := regexp.Compile(validServiceName)
+					if err != nil {
+						errs = append(errs, field.InternalError(path.Index(i).Child("method"),
+							fmt.Errorf("could not compile service name matching regex: %w", err)))
+					} else if !r.MatchString(*m.Method.Service) {
+						errs = append(errs, field.Invalid(path.Index(i).Child("method"), *m.Method.Service,
+							fmt.Sprintf("must only contain valid characters (matching %s)", validServiceName)))
```

Comment:
I tried to reword this but I think it may be easier to understand the regex than the equivalent in sentence form. Essentially we're saying that it must start with a letter or underscore, and then be followed with one or more letters, numbers, or underscores, plus a `.` is valid after at least two of those characters. I think that's actually missing some important nuance from the regex itself (`^(?i)\.?[a-z_][a-z_0-9]*(\.[a-z_][a-z_0-9]*)*$`), so I think it may be best to leave this as is.

Reply:
Updated in #1972.